sklearn/ensemble/base.py

```diff
@@ -178,3 +182,92 @@ def _partition_estimators(n_estimators, n_jobs):
     starts = np.cumsum(n_estimators_per_job)
 
     return n_jobs, n_estimators_per_job.tolist(), [0] + starts.tolist()
+
+
+class _BaseHeterogeneousEnsemble(MetaEstimatorMixin, _BaseComposition,
+                                 metaclass=ABCMeta):
+    """Base class for ensemble learners based on heterogeneous estimators.
```

Comment:
It's not the estimators that are heterogeneous right? It's the ensemble of estimator as far as I understand?